### PR TITLE
Include syntax vim generator

### LINKED ISSUE #28: Include with Vim
Hi, 
I'd like to re-use the vim-script generator and include it with Vim https://github.com/vim/vim/issues/13939. However, I see quite a few differences when running the Vim script syntax tests. I see the following failures:

It highlights (which upstream vim does not):
- `:Next`
- `:Print`
- `:X`

It doesn't highlight (which upstream vim does):

- `2match`
- `3match`
- `abbreviate`
- `augroup`
- `behave`
- `cabbrev`
- `cnoreabbrev`
- `cunmap`
- `echohl`
- `execute`
- `filetype`
- `global/.../` (the `/` delimiters), same for `:vglobal`
- `iabbrev`
- `inoreabbrev`
- `iunmap`
- `lunmap`
- `noreabbrev`
- `normal`
- `nunmap`
- `ounmap`
- `sunmap`
- `tunmap`
- `unmap`
- `vunmap`
- `xunmap`
- String interpolation `$"...."` (the `$`)
- Hex, Unicode and Octal, Key Codes (e.g. `\XF - \u02a4`) it highlights jus the `\X` while Vim highlights the whole word
- `<NL>`, `<NewLine>`, `<Enter>`, `<CSI>`, `<xCSI>`, `<PasteStart>`, `<PasteEnd>`, `<kInsert>`, `<zHome>`, `<zEnd>`
- `<NetMouse>`, `<DecMouse>`, `<JsbMouse>`, `<PtermMouse>`, `<UrxvtMouse>`, `<SgrMouse>`, `<SgrMouseRelease>`, `<LeftReleaseNM>`, `<MouseMove>`, `<ScrollWheelUp>`, `<ScrollWheelDown>`, `<ScrollWheelRight>`, `<ScrollWheelLeft>`, `<MouseDown>`, `<MouseUp>`, `<X1Mouse>`, `<X1Drag>`, `<X1Release>`, `<X2Mouse>`, `<X2Drag>`, `<X2Release>`, `<Drop>`
- `<SNR>`, `<CursorHold>`, `<ScriptCmd>`, `<FocusGained>`, `<FocusLost>`, `<stack>`, `<sflnum>`
- `<script>` is highlighted differently than Vim
- Line continuation: `"\ ...."`
- `conceal`
- `cchar=..`
- `contained`
- `containedin`
- `nextgroup`
- `transparent`
- `skipwhite`
- `skipempty`
- `skipnl`
- `contains`
- `oneline`
- `extend`
- `concealends`
- `excludenl`
- `keepend`
- `add/remove`
- `%=`, `/=`, `*=`
- register names `@f = `
- option names `let &g:foo = `
- List unpack: `let [foo, bar] .= expr`
- `unlet foo bar`
- `const foo = expr`

You can run the Vim Syntax scripts and see the results like this: 
```
cd ~/code/vim/runtime/syntax/
make
cat testdir/messages
for i in testdir/failed/*.dump; do vim --clean -c "call term_dumpdiff(\"testdir/failed/${i##*/}\", \"testdir/dumps/${i##*/}\")"; done
```

Do you think, this can be fixed? What do you think?

### LINKED ISSUE #13939: A vim.vim syntax script
Hi,
we have been struggeling in getting up to date versions for the Vim syntax script. I finally got the vim syntax script generator from Charles, but it is full of custom commands using different regex patterns that I cannot easily convert it into something useable. At least not without spending several days on it and I cannot even verify using the original tooling. 

The original is here: https://gist.github.com/chrisbra/58a5e36a8ab4fd41cf973b934a487c9b It contains several references to a grep like tool called xgrep and change which is similar to sed, but uses slightly different regex patterns and there are other references like `Uniq` and `mcol` comamnds, that I don't know what they do. There is also a mkvimvim.vim syntax script referenced, which I don't have. 

I heard that the vim-jp community has a syntax-script generator available here: https://github.com/vim-jp/syntax-vim-ex/tree/generator

Is this something we can use to re-generate the vim syntax scripts? How well does this work?

@k-takata @mattn Any insights? Can we integrate it into the Vim code tree, so generating the syntax is easier? Does it support Vim9 Script?

### PR DESCRIPTION
**Pull request by h-east**
Close #13939 
Close https://github.com/vim-jp/syntax-vim-ex/issues/28

I added runtime/syntax/generator/*

Generate runtime/syntax/vim.vim by performing the following operations.
```bash
$ cd /path-to-your-vim/runtime/syntax/generator
$ make
```

NOTE:
- After this PR is merged, please change the syntax logic to `runtime/syntax/generator/vim.vim.base` instead of `runtime/syntax/vim.vim`.
- `runtime/syntax/generator/vim.vim.base` reflects the latest changes in `runtime/syntax/vim.vim` (https://github.com/vim/vim/commit/b614b284ee06523511308f8381b47db34d921d39).

### CODE REVIEW

**Comment by pheiduck** [comment]
@h-east you missed to add 
```
-runtime/syntax/generator/Makefile
-runtime/syntax/generator/README.md
-runtime/syntax/generator/gen_syntax_vim.vim
-runtime/syntax/generator/update_date.vim
-runtime/syntax/generator/vim.vim.base
```
to the Filelist ;)

**Comment by chrisbra** [comment]
wow great. Does it pass the syntax tests?

**Comment by h-east** [comment]
Yes :+1:

**Comment by chrisbra** [comment]
@dkearns opinions ?

**Comment by chrisbra** [comment]
Thanks all, appreciate the team-work!

**Comment by zeertzjq** [comment]
`LineNrAbove` and `LineNrBelow` seems missing from the highlight groups. They were previously included.

**Comment by h-east** [comment]
I checked.
Missing `MessageWindow`, `Terminal` highlight group too.

**Comment by h-east** [comment]
> Missing `MessageWindow`, `Terminal` highlight group too.

This was wrong in the original. I don't think such a highlight group exists.

File: runtime/syntax/vim.vim [hunk 1]
```diff
@@ -35,39 +43,54 @@ syn keyword vimTodo contained	COMBAK	FIXME	TODO	XXX
...
+syn keyword vimCommand contained stat[ic] stopi[nsert] stj[ump] sts[elect] sun[hide] sus[pend] sv[iew] sw[apname] synti[me] sync[bind] smi[le] t tN[ext] ta[g] tags tab tabc[lose] tabd[o] tabe[dit] tabf[ind] tabfir[st] tabm[ove] tabl[ast] tabn[ext] tabnew tabo[nly] tabp[revious] tabN[ext] tabr[ewind] tabs tc[d] tch[dir] tcl tcld[o] tclf[ile] te[aroff] ter[minal] tf[irst] th[row] thi[s] tj[ump] tl[ast] tm[enu] tn[ext] to[pleft] tp[revious] tr[ewind] try ts[elect] tu[nmenu] ty[pe] u[ndo] undoj[oin] undol[ist] una[bbreviate] unh[ide] unl[et] unlo[ckvar] uns[ilent] up[date] v[global] ve[rsion] verb[ose] vert[ical] vi[sual] vie[w] vim[grep] vimgrepa[dd] vim9[cmd] vim9s[cript] viu[sage] vne[w] vs[plit] w[rite] wN[ext] wa[ll] wh[ile] wi[nsize] winc[md] wind[o] winp[os]
+syn keyword vimCommand contained wn[ext] wp[revious] wq wqa[ll] wu[ndo] wv[iminfo] x[it] xa[ll] xr[estore] y[ank] z dl dell delel deletl deletel dp dep delp delep deletp deletep a a
```
Comment (dkearns) [inline runtime/syntax/vim.vim:53]: These `a` commands are presumably an error - `append` is listed as an exclusion and they're duplicated.

Comment (h-east) [reply runtime/syntax/vim.vim:53]: Fixed: https://github.com/vim/vim/pull/14031/commits/463725e8a670487da4211f6a6dedc52116429ec5

**Comment by h-east** [comment]
@zeertzjq 
> LineNrAbove and LineNrBelow seem missing from the highlight groups. They were previously included.

Fixed: https://github.com/vim/vim/pull/14031/commits/8ded7a2dc0e112cebb81fbbd1e7b2a699686f86a

**Pull request closed without merging**